packages/react-components/src/components/StartCaptionsButton.tsx

```diff
@@ -93,16 +93,25 @@ export const _StartCaptionsButton = (props: _StartCaptionsButtonProps): JSX.Elem
     return { spokenLanguage: currentSpokenLanguage === '' ? defaultSpokenLanguage : currentSpokenLanguage };
   }, [currentSpokenLanguage]);
 
+  const [hasSetSpokenLanguage, setHasSetSpokenLanguage] = useState(false);
+
   const onToggleStartCaptions = useCallback(async (): Promise<void> => {
     if (props.checked) {
       onStopCaptions();
     } else {
       await onStartCaptions(options);
+    }
```

Comment:
This code is kinda duplicated for 2 components - I guess we just need it living in one place then we should be good? A place that wouldn't be disposed during the composite life cycle

Reply:
Cause it is not depending on the click of the button anymore - it only depends on setHasSetSpokenLanguage and isCaptionsFeatureActive